benchmarks/hash_table/dynamic_map_bench.cu

```diff
@@ -64,11 +34,12 @@ static void BM_dynamic_insert(::benchmark::State& state)
 
   std::size_t num_keys     = state.range(0);
   std::size_t initial_size = 1 << 27;
+  std::size_t multiplicity = 1;
```

Comment:
```suggestion
  constexpr std::size_t multiplicity = 1;
```
Use `const` or `constexpr` when possible. Same for all changes involved in this PR.